test/IntegrationTests/MSTest.Acceptance.IntegrationTests/AnalyzersTests.cs

```diff
@@ -116,26 +122,23 @@ public async void TestMethod1()
     }
 }
 """.PatchTargetFrameworks(TargetFrameworks.NetCurrent)
-    .PatchCodeWithReplace("$MSTestVersion$", MSTestVersion);
+    .PatchCodeWithReplace("$MSTestVersion$", MSTestVersion)
+    .PatchCodeWithReplace("$MSTestAnalysisMode$", analysisMode);
 
         // MSTEST0003 is TestMethodShouldBeValidAnalyzer, which is escalated to error in Recommended and All.
         // MSTEST0004 is PublicTypeShouldBeTestClassAnalyzer. Info and not enabled by default.
         // MSTEST0014 is DataRowShouldBeValidAnalyzer. Warn and enabled by default.
         // MSTEST0017 is AssertionArgsShouldBePassedInCorrectOrder. Info and enabled by default.
         // MSTEST0021 is PreferDisposeOverTestCleanupAnalyzer, which is disabled even in All mode.
         using TestAsset testAsset = await TestAsset.GenerateAssetAsync(nameof(VerifyMSTestAnalysisModeForDifferentAnalyzers), code);
-        await AssertAnalysisModeAsync("None", contains: [], doesNotContain: ["MSTEST0003", "MSTEST0004", "MSTEST0014", "MSTEST0017", "MSTEST0021"], testAsset.TargetAssetPath);
-        await AssertAnalysisModeAsync(string.Empty, contains: ["warning MSTEST0003", "warning MSTEST0014"], doesNotContain: ["MSTEST0004", "MSTEST0017", "MSTEST0021"], testAsset.TargetAssetPath);
-        await AssertAnalysisModeAsync("Default", contains: ["warning MSTEST0003", "warning MSTEST0014"], doesNotContain: ["MSTEST0004", "MSTEST0017", "MSTEST0021"], testAsset.TargetAssetPath);
-        await AssertAnalysisModeAsync("Recommended", contains: ["error MSTEST0003", "warning MSTEST0014", "warning MSTEST0017"], doesNotContain: ["MSTEST0004", "MSTEST0021"], testAsset.TargetAssetPath);
-        await AssertAnalysisModeAsync("All", contains: ["error MSTEST0003", "warning MSTEST0004", "warning MSTEST0014", "warning MSTEST0017"], doesNotContain: ["MSTEST0021"], testAsset.TargetAssetPath);
+        await AssertAnalysisModeAsync(analysisMode, contains, doesNotContain, testAsset.TargetAssetPath);
     }
 
     private static async Task AssertAnalysisModeAsync(string mode, string[] contains, string[] doesNotContain, string targetAssetPath)
     {
         // --no-incremental is due to https://github.com/dotnet/sdk/issues/46133.
         // Not sure if it's worth trying to find a workaround for it.
-        async Task<DotnetMuxerResult> BuildTaskAsync() => await DotnetCli.RunAsync($"build {targetAssetPath} -p:MSTestAnalysisMode={mode} --no-incremental", AcceptanceFixture.NuGetGlobalPackagesFolder.Path, warnAsError: false, retryCount: 0);
+        async Task<DotnetMuxerResult> BuildTaskAsync() => await DotnetCli.RunAsync($"build {targetAssetPath} --no-incremental", AcceptanceFixture.NuGetGlobalPackagesFolder.Path, warnAsError: false, retryCount: 0);
```

Comment:
`--no-incremental` is probably no longer needed as you generate new asset now for each case under test?